### ISSUE TITLE
filter function on the history page does not work

### ISSUE THREAD

**Issue by csecsi** [StackStorm/st2web#925]
## SUMMARY

the filter function on the history page does not do anything

### STACKSTORM VERSION

st2 3.6.0, on Python 3.6.8

##### OS, environment, install method

Centos 8 / ansible-st2

## Steps to reproduce the problem

just try to filter the history

## Expected Results

filtering the history

## Actual Results

not filtering the history

Thanks!

**Comment by amanda11** [comment]
Transferred to st2web repo as this is a UI issue.

**Comment by amanda11** [comment]
I have not seen this problem before, but I will re-do an installation today.
In the meantime could you:
- ensure st2web has fully loaded the history page. Just wondering if you have a large history and its still loading when you attempt to filter?
- using the developer tools, take a look at the API calls made during the filter calls and see if any errors reported
- are there any errors reported on developer tools in the console?
- which options have you tried to use on the filter?
- when you change the filter can you take a look at the API calls that were made - you should see the API call being made to the filter and the response coming back from the developer tools, Network tab.

**Comment by amanda11** [comment]
@csecsi If you could check the above that would be great. I performed a fresh install, and the filter on the UI is working fine for me on the history.

**Comment by csecsi** [comment]
![Selection_354](https://user-images.githubusercontent.com/11178380/143084933-021d673d-31f6-43a8-a3a3-c0146f507e5e.jpg)

Is it possible that I do not understand how to use the filter??? :)
It is on a fresh install.

**Comment by amanda11** [comment]
Apologies - I was thinking you were talking about the pull-downs on the History page, whereby you can filter on Action etc. 

I have the same behaviour if I use the free-form Filter, but I don't tend to use that. So I am not sure if that is due to format of the query or not at this stage.

**Comment by csecsi** [comment]
Weird that there is nothing logged when used. 
Is it possible that this feature is not even implemented?

**Comment by cognifloyd** [comment]
I see this behavior with:
- the unstable 3.7.0-dev packages.
- stable 3.6.0
- stable 3.4.1

**Comment by david-grokstream** [comment]
Is there any plan on fixing this?  It has been around for a while and we are trying to figure out if there is a workaround.

Ideally having a way of searching values in the action inputs would be handy (looking for hostnames, ticket ID, etc)

**Comment by csecsi** [comment]
looks like it is just you and me would need that feature working. :)

**Comment by fdrab** [comment]
> looks like it is just you and me would need that feature working. :)

Not true, but it's possible that companies are creating their own UIs or their own modifications to this UI, which then becomes company intellectual property. For example, I've added a time range filter, a filter for a specific execution ID (text input) and changed how the filter behaves by adding a "filter" button and now instead of there being an API call done whenever I'd change one of the values, I change the values and then click "filter" to filter executions.

**Comment by fdrab** [comment]
> Is there any plan on fixing this? It has been around for a while and we are trying to figure out if there is a workaround.
> 
> Ideally having a way of searching values in the action inputs would be handy (looking for hostnames, ticket ID, etc)

I don't think this is currently possible, because the API doesn't support searching by parameters, as far as I know. And the UI is just a component on top of the API. Action parameters can be complex objects and are not searchable. It is best if you create some sort of action reporting, like add action results into a SQL DB and then you can look up the execution by the parameters you want and look up a specific execution ID in ST2.
